### PR TITLE
Feature/apigateway authorization scopes

### LINKED ISSUE #4532: api_gateway_method authorization_scopes
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

It shall be possible to set "authorization_scopes" inside "api_gateway_method" when "authorization" is set to "COGNITO_USER_POOLS"

### New or Affected Resource(s)

* aws_api_gateway_method

### References

* #0000

### PR DESCRIPTION
**Pull request by tomjanne**
Fixes #4532

### CODE REVIEW

File: aws/resource_aws_api_gateway_method.go [hunk 1]
```diff
@@ -51,6 +51,12 @@ func resourceAwsApiGatewayMethod() *schema.Resource {
 				Optional: true,
 			},
 
+			"authorization_scopes": &schema.Schema{
+				Type:     schema.TypeList,
```
Comment (bflad) [inline aws/resource_aws_api_gateway_method.go]: Does the ordering matter for this attribute? If not, we can use `schema.TypeSet` which has some nice helpers for doing set differences. See also: https://www.terraform.io/docs/extend/schemas/schema-types.html#typeset

File: aws/resource_aws_api_gateway_method.go [hunk 2]
```diff
@@ -165,6 +175,7 @@ func resourceAwsApiGatewayMethodRead(d *schema.ResourceData, meta interface{}) e
 	d.Set("api_key_required", out.ApiKeyRequired)
 	d.Set("authorization", out.AuthorizationType)
 	d.Set("authorizer_id", out.AuthorizerId)
+	d.Set("authorization_scopes", out.AuthorizationScopes)
```
Comment (bflad) [inline aws/resource_aws_api_gateway_method.go]: When setting non-scalar attributes, we should perform error checking:

```go
if err := d.Set("authorization_scopes", out.AUthorizationScopes); err != nil {
  return fmt.Errorf("error setting authorization_scopes: %s", err)
}
```

File: aws/resource_aws_api_gateway_method.go [hunk 3]
```diff
@@ -229,6 +240,35 @@ func resourceAwsApiGatewayMethodUpdate(d *schema.ResourceData, meta interface{})
...
 	}
 
+	if d.HasChange("authorization_scopes") {
+		o, n := d.GetChange("authorization_scopes")
+		path := "/authorizationScopes"
+
+		old := o.([]interface{})
+		new := n.([]interface{})
+
+		// Remove every authorization scope. Simpler to remove and add new ones,
```
Comment (bflad) [inline aws/resource_aws_api_gateway_method.go]: Note here: If we switch this to `schema.TypeSet`, there are some set difference functions available: https://godoc.org/github.com/hashicorp/terraform/helper/schema#Set.Difference

It seems like we should try to reduce any downtime associated with updating these if possible.

**Comment by bflad** [comment]
@tomjanne did you see the feedback above or have any questions?

**Comment by tomjanne** [comment]
@bflad sorry for the delay. i am currently working on a new version. Maybe i can update the pull-request today/tomorrow.

**Comment by tomjanne** [comment]
Output from Acceptance-Test:

`make testacc TESTARGS='-run=TestAccAWSAPIGatewayMethod_cognitoauthorizer'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v -run=TestAccAWSAPIGatewayMethod_cognitoauthorizer -timeout 120m
?   	github.com/terraform-providers/terraform-provider-aws	[no test files]
=== RUN   TestAccAWSAPIGatewayMethod_cognitoauthorizer
--- PASS: TestAccAWSAPIGatewayMethod_cognitoauthorizer (63.56s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	63.655s`

**Comment by bflad** [comment]
This has been released in [version 1.22.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1220-june-05-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Pull request merged**